cabal-docspec/MANUAL.md

```diff
@@ -523,6 +523,101 @@ Failures in the setup code does not cause module skip.
 
 GHC-7.0 relies that *Char* type is in scope. This is an implementation artifact.
 
+Q&A
+====
+
+Q: cabal-docspec doesn't import modules automatically
+-----------------------------------------------------
+
+cabal-docspec tests library documentation from the outside.
+It doesn't even try to look into an implementation for some secret bits, only to find examples.
+In this sense it is more principled (than doctest).
+There fore you might need to repeat imports in **$setup** block.
+OTOH, the implementation imports never interfere with doctests.
+
+Q: How to hide some Prelude imports, e.g. null
+----------------------------------------------
+
+One way is to redefine the symbol in **$setup** block using qualified
+module name.
+
+```haskell
+let null = Module.Under.Test.null
+```
+
+This way it will shadow both *Prelude.null* and *Module.Under.Test.null*,
+and ambiguous symbol errors wont appear.
+
+Another option is to use **-XNoImplicitPrelude** and import *Prelude*
+explicitly.
+
+Q: How cabal-docspec works with pre-8.0 version of GHC
+------------------------------------------------------
+
+cabal-docspec reads *plan.json* file which is generated by cabal-install.
+That file contains (almost) all required information for cabal-docspec
+to invoke ghci with the correct arguments.
+
+Q: When does plan.json get generated?
+-------------------------------------
+
+It is generated by cabal-install as a side-effect of running the solver.
+For example even
+
+    cabal build --dry-run
+
+is enough.
+However, without libraries actually being built, cabal-docspec won't work.
+
+Q: Does doctests' --fast have an equivalent in cabal-docspec?
+-------------------------------------------------------------
+
+No, cabal-doctest doesn't need one.
+As the library under tests is not interpreted, but loaded from
+pre-compiled object, the **:reload** command is cheap.
+It doesn't cause the re-interpretation of the sources.
+
+Q: Are you envisioning making binary distributions of cabal-docspec available
+-----------------------------------------------------------------------------
+
+Yes.
+
+Q: In the lens example, is test-suite somehow related to doctests?
```

Comment:
Perhaps it would be best just to include this information in section about the `lens` example, rather than including it here (which is relatively far away).